apps/crn-server/src/data-providers/contentful/guides.data-provider.ts

```diff
@@ -0,0 +1,47 @@
+import { GuideDataObject, ListGuideResponse } from '@asap-hub/model';
+import {
+  GraphQLClient,
+  FETCH_GUIDE,
+  FetchGuideQuery,
+  GuideContent,
+} from '@asap-hub/contentful';
+import { GuideDataProvider } from '../types';
+
+type GuideItem = NonNullable<
+  NonNullable<FetchGuideQuery['guidesCollection']>['items']
+>[number];
+
+type GuideContentItem = NonNullable<
+  Pick<GuideContent, 'text' | 'title' | 'linkUrl' | 'linkText'>[]
+>[number];
+
+export class GuideContentfulDataProvider implements GuideDataProvider {
+  constructor(private contentfulClient: GraphQLClient) {}
+
+  async fetch(): Promise<ListGuideResponse> {
+    const { guidesCollection } =
+      await this.contentfulClient.request<FetchGuideQuery>(FETCH_GUIDE);
+    return {
+      items: parseGraphQLGuides(guidesCollection?.items || []),
+      total: guidesCollection?.items.length || 0,
+    };
+  }
+}
+const parseGuideContent = (guideContent: GuideContentItem | null) => ({
+  text: guideContent?.text || '',
+  title: guideContent?.title || '',
+  linkUrl: guideContent?.linkUrl || '',
+  linkText: guideContent?.linkText || '',
+});
+
+export const parseGraphQLGuide = (guide: GuideItem): GuideDataObject => ({
+  title: guide?.title ? guide.title : '',
+  content: guide?.contentCollection
+    ? guide.contentCollection.items.map((content: GuideContentItem | null) =>
+        parseGuideContent(content),
+      )
+    : [],
+});
```

Comment:
I think you'll want to remove any null items from the response rather than populating with all empty strings.

Reply:
that's actually debatable, for some reason this has been the way we have done things in CRN although I would say that unless its a single value normalised to an empty string it should throw some sort of exception (As in when all values are undefined we should throw)